src/test/java/com/puppycrawl/tools/checkstyle/utils/CodePointUtilTest.java

```diff
@@ -0,0 +1,130 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2021 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.utils;
+
+import static com.google.common.truth.Truth.assertWithMessage;
+import static com.puppycrawl.tools.checkstyle.internal.utils.TestUtil.isUtilsClassHasPrivateConstructor;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+
+import java.io.File;
+import java.nio.charset.Charset;
+
+import org.junit.jupiter.api.Test;
+
+import com.puppycrawl.tools.checkstyle.AbstractPathTestSupport;
+import com.puppycrawl.tools.checkstyle.api.FileContents;
+import com.puppycrawl.tools.checkstyle.api.FileText;
+import com.puppycrawl.tools.checkstyle.checks.AvoidEscapedUnicodeCharactersCheck;
+
+public class CodePointUtilTest extends AbstractPathTestSupport {
```

Comment:
Please avoid writing synthetic tests for utility classes.

Just add AvoidEscapedUnicodeCharactersCheckTest to the list of  targets:
https://github.com/checkstyle/checkstyle/blob/abb48248bdf3aade4e8972d7231b0b5969e8d557/pom.xml#L3272

Reply:
> Please avoid writing synthetic tests for utility classes.

https://github.com/checkstyle/checkstyle/pull/11094#discussion_r775049091

Reply:
Can I just use "somestring".codepoints().toArray() as inputs for test in CodePointUtil?

Reply:
> "somestring".codepoints().toArray()

This is also "synthetic", we try to only have tests on real input that integrates as much of the project as we can. For a util class, it should prove it's "utility" by being able to have all code/ mutation coverage come from actual usages. If not, then why should such code exist?

> Just add AvoidEscapedUnicodeCharactersCheckTest to the list of targets

@pbludov I agree, this is the best way. However, we may end up adding 4-5 more target tests to `pitest-utils` by the time this issue has been completed. Are we good with this? I thought I remembered from somewhere that we were trying to limit number of target tests in pitest config.

Reply:
@nmancus1 So what I assume is 
I should remove `CodePointUtilTest` and add affected check (`AvoidEscapedUnicodeCharactersCheckTest`) in `targetTests`. Am I correct?
(Sorry, I have been taking too much time, since this is one of the initial issues with this project).

But I see `CommonUtilTest` uses string inputs for isBlank, isInt.

Reply:
@MUzairS15 yes, remove `CodePointUtilTest`. We will use `AvoidEscapedUnicodeCharactersCheckTest` to cover `CodePointUtil` in `pitest-utils`, per @pbludov. 

> Sorry, I have been taking too much time...

No worries, since this is the first such PR (for #10924), it is important that we get expectations for future updates figured out.

> But I see CommonUtilTest uses string inputs for isBlank, isInt...

Yes, this is an area that we are constantly trying to improve in. There are a lot of "synthetic" tests in the project, we are trying to eliminate adding any more of them.

Synthetic tests do not prove that code works well in practice, they only prove that code works as engineer wrote it :)

Reply:
Thankyou, for your response.

Reply:
@pbludov we must keep synthetic test for private constructor: https://github.com/checkstyle/checkstyle/runs/4643851189?check_suite_focus=true

> remove `CodePointUtilTest`

@MUzairS15 this was a mistake on my part, we must keep at least this one synthetic test for private constructor to have 100%  mutation coverage.

Reply:
done

Reply:
> @pbludov I agree, this is the best way. However, we may end up adding 4-5 more target tests to `pitest-utils` by the time this issue has been completed. Are we good with this? I thought I remembered from somewhere that we were trying to limit number of target tests in pitest config.

If the pitest profile for utility classes becomes too large, we can split it.